src/main.rs

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) 2022 Jan Holthuis <jan.holthuis@rub.de>
+//
+// This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0. If a copy
+// of the MPL was not distributed with this file, You can obtain one at
+// http://mozilla.org/MPL/2.0/.
+//
+// SPDX-License-Identifier: MPL-2.0
+
+use binrw::{BinRead, ReadOptions};
+use clap::{Parser, Subcommand};
+use rekordcrate::anlz::ANLZ;
+use rekordcrate::pdb::{Header, PageType, Row};
+use rekordcrate::setting::Setting;
+use std::path::PathBuf;
+
+#[derive(Parser)]
+#[command(author, version, about)]
+#[command(propagate_version = true)]
+struct Cli {
+    #[command(subcommand)]
+    command: Commands,
+}
+
+#[derive(Subcommand)]
+enum Commands {
+    /// List the playlist tree from a Pioneer Database (`.PDB`) file.
+    ListPlaylists {
+        /// File to parse.
+        #[arg(value_name = "PDB_FILE")]
+        path: PathBuf,
+    },
+    /// Parse and dump a Rekordbox Analysis (`ANLZXXXX.DAT`) file.
+    DumpANLZ {
+        /// File to parse.
+        #[arg(value_name = "ANLZ_FILE")]
+        path: PathBuf,
+    },
+    /// Parse and dump a Pioneer Database (`.PDB`) file.
+    DumpPDB {
+        /// File to parse.
+        #[arg(value_name = "PDB_FILE")]
+        path: PathBuf,
+    },
+    /// Parse and dump a Pioneer Settings (`*SETTING.DAT`) file.
+    DumpSetting {
+        /// File to parse.
+        #[arg(value_name = "SETTING_FILE")]
+        path: PathBuf,
+    },
+}
+
+fn list_playlists(path: &PathBuf) {
+    use rekordcrate::pdb::PlaylistTreeNode;
+    use std::collections::HashMap;
+
+    fn print_children_of(tree: &HashMap<u32, Vec<PlaylistTreeNode>>, id: &u32, level: usize) {
```

Comment:
```suggestion
    fn print_children_of(tree: &HashMap<u32, Vec<PlaylistTreeNode>>, id: u32, level: usize) {
```
why pass the `Copy` type as ref?

Reply:
https://github.com/Holzhaus/rekordcrate/pull/82/commits/1fd43c118553d1ff18abe52b3bb8272069d69aea